### PR TITLE
Fixed compilation with deprecated input/output methods in Node

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: ngraph/core/include/openvino/core/node.hpp [hunk 1]
```diff
@@ -305,75 +304,139 @@ class OPENVINO_API Node : public std::enable_shared_from_this<Node> {
...
+
+    const Shape& input_shape(size_t i) const;
+    const element::Type& input_element_type(size_t i) const;
+    descriptor::Tensor& input_tensor(size_t i) const;
+    Node* input_node_ptr(size_t index) const;
+    std::shared_ptr<Node> input_node_shared_ptr(size_t index) const;
+    Output<Node> input_source_output(size_t i) const;
+    /// Returns the tensor for nput i
+    OPENVINO_DEPRECATED("The method was deprecated! Please use input_tensor() instead.")
+    descriptor::Tensor& get_input_tensor(size_t i) const;
```
Comment (ilya-lavrenov) [inline ngraph/core/include/openvino/core/node.hpp:411]: it looks too much changes which don't bring any benefits

Comment (ilyachur) [reply ngraph/core/include/openvino/core/node.hpp:411]: It allows to align the function names.
From my point of view it is strange to have a mix of methods:
```
outputs();
output();
inputs();
input();
input_shape();
output_shape();

// and
get_input_element_type()
get_input_tensor()
// etc.
```

Comment (ilya-lavrenov) [reply ngraph/core/include/openvino/core/node.hpp:411]: > It allows to align the function names.

Which one? Please, provide a description what we are trying to do

Comment (ilyachur) [reply ngraph/core/include/openvino/core/node.hpp:411]: Let's discuss it offline

**Pull request closed without merging**